platforms/AndroidFireTV/RefApp/app/src/main/java/com/truex/ctvwebrefapp/MainActivity.kt

```diff
@@ -57,8 +59,39 @@ class MainActivity : Activity() {
         webSettings.setAppCacheEnabled(false)
         webSettings.cacheMode = WebSettings.LOAD_NO_CACHE
 
-        val appUrl = getString(R.string.app_url);
+        webView.webViewClient = object : WebViewClient() {
+            override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+                return false
+            }
+
+            override fun onReceivedError(view: WebView?, request: WebResourceRequest?, error: WebResourceError?) {
+                super.onReceivedError(view, request, error)
+                hideSplashScreen()
+            }
+
+            override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+                super.onPageStarted(view, url, favicon)
+                // Ensure the splash screen gets removed eventually, e.g. if the webview fails to load.
+                Handler().postDelayed(Runnable {
+                    hideSplashScreen()
+                }, 3000)
+            }
+        }
+
+        val appUrl = getString(R.string.app_url)
         webView.loadUrl(appUrl)
+
+    }
+
+    @JavascriptInterface
+    fun hideSplashScreen() {
+        runOnUiThread {
+            var mainLayout: ViewGroup = findViewById(R.id.mainLayout)
+            var splashScreen: View? = findViewById(R.id.appSplash)
+            if (splashScreen != null && splashScreen.parent != null) {
+                mainLayout.removeView(splashScreen)
```

Comment:
Same comment here about mainLayout guard.  Although this would mean its new as opposed to old code too.